### ISSUE TITLE
Problem making interdependent protos

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#4137]
Looks like our makefile has trouble generating protos that depend on each other.

For example, test/proto/benchmarks/control.proto   imports   payloads.proto. So payloads.pb.h is included by control.pb.cc, but there is no way of expressing that payloads.proto  produces payloads.pb.h and that control.pb.cc depends on it.

That results in failure to build some targets that use multiple interdependent proto files (like those in test/proto/benchmarks).

```
$ make qps_driver
[CXX]     Compiling /usr/local/google/home/jtattermusch/github/grpc/gens/test/proto/benchmarks/control.pb.cc
In file included from /usr/local/google/home/jtattermusch/github/grpc/gens/test/proto/benchmarks/control.pb.cc:5:0:
/usr/local/google/home/jtattermusch/github/grpc/gens/test/proto/benchmarks/control.pb.h:31:47: fatal error: test/proto/benchmarks/payloads.pb.h: No such file or directory
 #include "test/proto/benchmarks/payloads.pb.h"
                                               ^
compilation terminated.
make: *** [/usr/local/google/home/jtattermusch/github/grpc/objs/opt//usr/local/google/home/jtattermusch/github/grpc/gens/test/proto/benchmarks/control.pb.o] Error 1
```

**Comment by jtattermusch** [comment]
CC @nicolasnoble @vjpai

**Comment by jtattermusch** [comment]
@ctiller , feel free to reassign.

**Comment by vjpai** [comment]
The contents of test/proto/benchmarks are not interdependent, though. 
<pre>
services --> control --> payloads
                     --> stats
         --> ../messages
</pre>

There are dependences, of course, but they're all one-directional.

**Comment by vjpai** [comment]
But I guess my note doesn't actually address your point. Even with dependences there is a problem. I don't know why this doesn't show up in our builds in general. I can repro this bug perfectly by doing a full make, then deleting the relevant file:

rm gens/test/proto/benchmarks/payloads.pb.h

and then watching the build script fall over.

**Comment by ctiller** [comment]
Predicating the fix for this on #4123... Once that's done it becomes much
more practical to solve this with some build.yaml markup (perhaps
ultimately autogenerated).
